Source/DafnyRuntime/DafnyRuntime.cs

```diff
@@ -769,12 +780,16 @@ public abstract class Sequence<T>: ISequence<T>
     public ISequence<T> Take(BigInteger n) {
       return Take((long)n);
     }
-    public ISequence<T> Drop(long m) {
-      if (m == 0)
+    public ISequence<T> Drop(long m)
+    {
+      int elem = checked((int)m);
+      if (elem == 0)
         return this;
-      T[] a = new T[Elements.Length - m];
-      System.Array.Copy(Elements, m, a, 0, Elements.Length - m);
-      return new ArraySequence<T>(a);
+      T[] tmp = new T[ImmutableElements.Length - (int)m];
+      // Doing the copy directly is faster than calling ImmutableArray's RemoveRange method, which performs the
+      // copy step twice (since it removes a range from the, potential, middle of an array)
+      System.Array.Copy(ImmutableElements.ToArray(), m, tmp, 0, ImmutableElements.Length - (int)m);
```

Comment:
If the parameterized `ToArray()` I was wishing for on L747 exists, then it could be used here, too.

Reply:
Btw, L791 still seems to make two copies: first all elements are copied by `ToArray()` and then `System.Array.Copy` copies the desired elements.

Reply:
Will update.